engine/core/src/main/java/org/datacleaner/util/LabelUtils.java

```diff
@@ -96,13 +97,26 @@ public static String getLabel(ComponentJob job) {
      * 
      * @return
      */
-    public static String getLabel(ComponentJob job, boolean includeDescriptorName, boolean includeInputColumnNames,
+    public static String getLabel(ComponentJob job, boolean includeDescriptorName, boolean includeInputColumnNames, boolean includeRequirements) {
+        return getLabel(job, null, includeDescriptorName, includeInputColumnNames, includeRequirements);
+    }
+
+    public static String getLabel(ComponentJob job, AnalyzerResult result, boolean includeDescriptorName, boolean includeInputColumnNames,
             boolean includeRequirements) {
         final String jobName = job.getName();
         final StringBuilder label = new StringBuilder();
         if (Strings.isNullOrEmpty(jobName)) {
             ComponentDescriptor<?> descriptor = job.getDescriptor();
-            label.append(descriptor.getDisplayName());
+            String displayName = descriptor.getDisplayName();
+            if (result != null) {
+                if (result instanceof HasLabelAdvice) {
```

Comment:
I'm probably guilty as charged. That's what you get for trying to explain something, I better shut up next time ;-)

I think there are only 3 implementing classes where the name stuff is used/produced (one for each of Transformer, Filter and Analyzer).